lib/mayaUsd/fileio/orphanedNodesManager.cpp

```diff
@@ -269,47 +391,118 @@ bool OrphanedNodesManager::isOrphaned(const Ufe::Path& pulledPath) const
         // If the argument path has not been pulled, it can't be orphaned.
         return false;
     }
-    TF_VERIFY(trieNode->hasData());
+
+    if (!trieNode->hasData()) {
+        // If the argument path has not been pulled, it can't be orphaned.
+        return false;
+    }
+
     // If the pull parent is visible, the pulled path is not orphaned.
-    return !getVisibilityPlug(trieNode);
+    const auto& pullParentPath = trieNode->data().pulledParentPath;
+    MFnDagNode  fn(pullParentPath);
+    auto        visibilityPlug = fn.findPlug("visibility", /* tryNetworked */ true);
+    return !visibilityPlug.asBool();
 }
 
-/* static */
-bool OrphanedNodesManager::setVisibilityPlug(
-    const Ufe::TrieNode<PullVariantInfo>::Ptr& trieNode,
-    bool                                       visibility)
+namespace ufe {
+
+extern Ufe::Rtid g_MayaRtid;
+extern Ufe::Rtid g_USDRtid;
+
+} // namespace ufe
+
+namespace {
+
+Ufe::Path trieNodeToPullePrimUfePath(Ufe::TrieNode<PullVariantInfo>::Ptr trieNode)
 {
-    TF_VERIFY(trieNode->hasData());
-    const auto& pullParentPath = trieNode->data().dagPath;
-    MFnDagNode  fn(pullParentPath);
-    auto        visibilityPlug = fn.findPlug("visibility", /* tryNetworked */ true);
-    return (visibilityPlug.setBool(visibility) == MS::kSuccess);
+    const PullVariantInfo& variantInfo = trieNode->data();
```

Comment:
Currently we do not store the pulled prim path in the data because it is "implicitly" stored in the Trie "key". This key is spread over multiple Trie nodes. Furthermore, the Trie only keeps UFE components, not segments, so we need to rebuild the UFE path by adding segments at the proper place.

Reply:
O.K.